packages/drivers/odsp-driver/src/createFile.ts

```diff
@@ -233,27 +264,76 @@ export async function createNewEmptyFluidFile(
     });
 }
 
-export async function createNewFluidFileFromSummary(
-    getStorageToken: InstrumentedStorageTokenFetcher,
-    newFileInfo: INewFileInfo,
+export type CreateNewFluidContainerFromSummaryArgs = [
     logger: ITelemetryLogger,
     createNewSummary: ISummaryTree,
     epochTracker: EpochTracker,
     forceAccessTokenViaAuthorizationHeader: boolean,
+];
+
+export async function createNewFluidFileFromSummary(
+    getStorageToken: InstrumentedStorageTokenFetcher,
+    fileInfo: INewFileInfo,
+    ...args: CreateNewFluidContainerFromSummaryArgs
 ): Promise<ICreateFileResponse> {
-    const filePath = newFileInfo.filePath ? encodeURIComponent(`/${newFileInfo.filePath}`) : "";
-    const encodedFilename = encodeURIComponent(newFileInfo.filename);
-    const baseUrl =
-        `${getApiRoot(getOrigin(newFileInfo.siteUrl))}/drives/${newFileInfo.driveId}/items/root:` +
-        `${filePath}/${encodedFilename}`;
+    const content = await createNewFluidContainerFromSummary<ICreateFileResponse>(
+        getStorageToken,
+        fileInfo,
+        ...args
+    );
+
+    if (isNewFileInfo(fileInfo)) {
+        if (!content || !content.itemId) {
+            throw new NonRetryableError(
+                "ODSP CreateFile call returned no item ID",
+                DriverErrorType.incorrectServerResponse,
+                { driverVersion });
+        }
+    }
+
+    return content;
+}
+export async function createNewFluidContainerOnExistingFileFromSummary(
+    getStorageToken: InstrumentedStorageTokenFetcher,
+    fileInfo: IExistingFileInfo,
+    ...args: CreateNewFluidContainerFromSummaryArgs
+): Promise<void> {
+    await createNewFluidContainerFromSummary<unknown>(
+        getStorageToken,
+        fileInfo,
+        ...args
+    );
+
+    return;
+}
+
+async function createNewFluidContainerFromSummary<T>(
```

Comment:
we shouldn't use an unbound generic type like this, as it doesn't preserve any type safety, you are basically just casting as any

Reply:
I feel this is similar to low level APIs like fetchAndParseAsJSON which the consumer needs to specify the type to pre-anticipate the type of a HTTP response.
In this case, the type of response is scenario based and the consumer of the API still has to specify the type. (i.e., <T> is still better than <T = any>)
I can make this internal and create two functions simply passing around the arguments and return the specific types for the two different scenarios, but that feels like an overkill to me. Any suggestions?